packages/react-native/React/Fabric/Mounting/ComponentViews/Image/RCTImageComponentView.mm

```diff
@@ -136,7 +136,10 @@ - (void)didReceiveImage:(UIImage *)image metadata:(id)metadata fromObserver:(con
     return;
   }
 
-  static_cast<const ImageEventEmitter &>(*_eventEmitter).onLoad(_state->getData().getImageSource());
+  auto imageSource = _state->getData().getImageSource();
```

Comment:
For someone not super familiar with this code, does this mean that the `ImageSource` stored as state is incorrect? Wondering if this change should happen here when emitting, or if we are storing wrong information earlier.

Reply:
@NickGerleman Hi, `ImageSource` is used for layout, it's correct for the state. we misuse the info when emit load events. 
 
Android part:
https://github.com/facebook/react-native/blob/8b53d41a88c3c7c83183baa46be307102aa90d54/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/image/ReactImageView.java#L200-L205